### PR TITLE
added firebase configuration and dependencies

### PR DESCRIPTION
**Pull request by bhanu-priyaa**

### CODE REVIEW

File: pages/lib/useFirebaseAuth.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  const signedInWithEmailAndPassword = (email, password) =>
+    signInWithEmailAndPassword(auth,email, password);
+
+    const signedOut = () =>
+        signOut(auth).then(clear);
+// listen for Firebase state change
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth,authStateChanged);
+    return () => unsubscribe();
+  }, []);
```
Comment (sutyum) [inline pages/lib/useFirebaseAuth.js:42]: dependency list should not be empty and instead depend on `auth`

File: package.json [hunk 2]
```diff
@@ -15,11 +15,15 @@
         "@heroicons/react": "^1.0.5",
         "@types/jest": "^29.2.4",
         "@types/react-dom": "^18.0.9",
+        "firebase": "^9.15.0",
+        "firebase-admin": "^11.4.0",
```
Comment (sutyum) [inline package.json:19]: why is this needed?

File: pages/index.tsx [hunk 3]
```diff
@@ -38,7 +43,26 @@ const geojson: GeoJSON.FeatureCollection = {
...
 const Home: NextPage = () => {
     const isDrawerOpen = useStore((state) => state.isDrawerOpen);
+    const{authUser,loading} = useAuth();
+    const router = useRouter();
 
+    useEffect(() => {
+        if (!loading && !authUser){
+            router.push('/login');
+        }
+    //   const auth = initFirebase;
```
Comment (sutyum) [inline pages/index.tsx:53]: if this is not needed anymore, better to delete

File: pages/login/index.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
+import React,{useState} from "react";
+// import { signInWithEmailAndPassword } from 'firebase/auth';
```
Comment (sutyum) [inline pages/login/index.tsx:2]: remove if not needed anymore

File: pages/login/index.tsx [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+                    }
+                    type="password"
+                    autoComplete="current-password"
+                    required
+                    className="appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-sky-500 focus:border-sky-500 sm:text-sm"
+                  />
+                </div>
+              </div>
+
+              {/* <div className="flex items-center justify-between">
```
Comment (sutyum) [inline pages/login/index.tsx:95]: remove

File: pages/login/index.tsx [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+              <div className="relative">
+                <div className="absolute inset-0 flex items-center">
+                  <div className="w-full border-t border-gray-300" />
+                </div>
+                <div className="relative flex justify-center text-sm">
+                  {/* <span className="px-2 bg-white text-gray-500">Or continue with</span> */}
+                </div>
+              </div>
+                {/* <div>
+                  <a
```
Comment (sutyum) [inline pages/login/index.tsx:135]: remove comments

File: pages/login/index.tsx (hunk 5 above)
Comment (sutyum) [reply pages/login/index.tsx:95]: remove comments

**Review by sutyum — CHANGES_REQUESTED** [review]

File: next.config.js [hunk 7]
```diff
@@ -4,4 +4,16 @@ const nextConfig = {
   reactStrictMode: true
 };
 
-module.exports = (nextConfig);
+module.exports = {
```
Comment (sutyum) [inline next.config.js:7]: share a reference on why you have done it line this, earlier this was fine

File: pages/index.tsx [hunk 8]
```diff
@@ -10,6 +20,39 @@ import { AppLayout } from '../component/AppLayout';
...
 import { LocationMarkerIcon } from '@heroicons/react/solid';
 
+export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
+    try {
+      const cookies = nookies.get(ctx);
+      const token = await firebaseAdmin.auth().verifyIdToken(cookies.token);
+      const { uid, email } = token;
+  
+      // the user is authenticated!
+      // FETCH STUFF HERE
```
Comment (sutyum) [inline pages/index.tsx:30]: Is this needed?

**Review by sutyum — CHANGES_REQUESTED** [review]
I have glanced at the PR, and some requested changes

File: next.config.js (hunk 7 above)
Comment (bhanu-priyaa) [reply next.config.js:7]: https://nextjs.org/docs/api-reference/next.config.js/environment-variables
This is a reference to show the addition of enviroment variables in config file

File: pages/api/entry/{id}.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+    console.log("the value of id is: ",id)
+
+    try{
+        const doc = await firebaseAdmin.collection('lands').doc(id).get();
+        const doc2 = await firebaseClient.collection
+        if(!doc.exists){
+            res.status(400).end();
+        }
+        else{
```
[the remark this answers is not in the document]
Comment (bhanu-priyaa) [reply pages/api/entry/{id}.js:15]: resolved issue with else block

**Review by sutyum — APPROVED** [review]

**Pull request merged**